.github/workflows/build_publish_openmrs.yml

```diff
@@ -1,50 +1,30 @@
 name: Build and Publish OpenMRS
 on:
   push:
-    branches: [ master ]
+    branches:
+      - master
+      - 'release-*'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
     paths-ignore:
       - "**.md"
-      - ".github/workflows/build_publish_openmrs_freshDB.yml"
-      - ".github/workflows/build_publish_openmrs_demoDB.yml"
-      - "package/docker/demoDB/**"
-      - "package/docker/freshDB/**"
 
   repository_dispatch:
     types: ["default-config-trigger"]
 
 jobs:
-  get-version:
-    #This job reads the .appversion file and checks it against tags in the repo so that overlap builds don't occur.
-    #Also provides version as output for other jobs.
-    name: Get Version
-    runs-on: ubuntu-latest
-    outputs:
-      version: ${{ steps.get-version.outputs.version }}
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-      - name: Set Version Output
-        id: get-version
-        run: |
-          VERSION=$(cat package/.appversion)
-          if [ $(git tag -l "$VERSION") ]; then 
-            echo "Version $VERSION already released. Please update your version in package/.appversion"
-            exit 1
-          else
-            echo "Setting version $VERSION-${{github.run_number}}"
-            echo ::set-output name=version::$VERSION-${{github.run_number}}
-          fi
-
   docker-build-publish:
     name: Docker Build & Publish
     runs-on: ubuntu-latest
-    needs: [ get-version ]
     env:
-      ARTIFACT_VERSION: ${{ needs.get-version.outputs.version }}
+      HELM_CHART_PATH: package/helm/openmrs
     steps:
       - uses: actions/checkout@v2
+      - name: Set artifact version
+        run: |
+          wget -q https://raw.githubusercontent.com/Bahmni/bahmni-infra-utils/main/setArtifactVersion.sh && chmod +x setArtifactVersion.sh
+          ./setArtifactVersion.sh
+          rm setArtifactVersion.sh
```

Comment:
This is the right approach except that it would be eventually difficut for someone to understand that the script is setting the env.ARTIFACT_VERSION (side effect). So either you could make the script PURE by not creating any side effects and just return the artifact version number which can be forwarded to `>> $GITHUB_ENV` OR to least change the name to indicate its setting `env.ARTIFACT_VERSION`

Reply:
We have refactored the step name as `Set env.ARTIFACT_VERSION`